src/com/vaadin/plugin/CopilotRestService.java

```diff
@@ -0,0 +1,57 @@
+package com.vaadin.plugin;
+
+import com.sun.net.httpserver.HttpServer;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpExchange;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Minimal REST service for Vaadin Copilot integration. Starts on a random
+ * free port and exposes a single /copilot endpoint accepting POST requests.
+ */
+public class CopilotRestService {
+    private HttpServer server;
+    private int port;
+
+    public void start() throws IOException {
+        server = HttpServer.create(new InetSocketAddress(0), 0);
```

Comment:
Instead of creating a HttpServer here, can we use the existing one and register a servlet instead? Something like
```
    HttpService httpService = ...; // get from OSGi context
    httpService.registerServlet("/myapi", new MyServlet(), null, null);
```
?